scripts/bmm_examples/Example4_unpublished_monotonic_constraints.R

```diff
@@ -9,57 +9,88 @@
+if (nChains >  parallel::detectCores()) {
+  nChains <-  parallel::detectCores()
+}
+
+# set brms controls to prevent divergent transitions & improve convergence
+adapt_delta <- .95
+max_treedepth <- 10
+
 data_popov <- read.csv(here('data/popov_unpublished.csv'))
 data_popov$setsize <- as.factor(data_popov$setsize)
 
 #############################################################################!
 # 1) Model specification and estimation                                  ####
 #############################################################################!
 
+model_2p <- mixture2p(resp_error = "anglediff")
+
 # code contrasts such that the effect of each size is relative to the effect of 
 # the setsize before
-contr_ordin <- matrix(0, nrow=8, ncol=7)
+contr_ordin <- matrix(0, nrow = 8, ncol = 7)
 contr_ordin[lower.tri(contr_ordin)] <- -1
 colnames(contr_ordin) <- 2:8
 contrasts(data_popov$setsize) <- contr_ordin
 contrasts(data_popov$setsize)
 
-ff <- bf(anglediff ~ 1,
-         kappa ~ setsize,
-         thetat ~ setsize)
-
-pr <- prior_('normal(0.0,0.8)', class='b', nlpar='kappa', lb=0) +
-  prior_('logistic(0,1)', class='b', nlpar='thetat',lb=0)
-
-
-
-filename <- 'output/fit_popov_unpublished_monotonic_constraints_2p.RData'
-if (!file.exists(here(filename))) {
-  
-  fit_popov <- fit_model(formula = ff,
-                         data = data_popov,
-                         model_type = "2p",
-                         parallel = TRUE,
-                         warmup = 500,
-                         iter = 1000,
-                         prior=pr)
-  
-  save(list=ls(), file=here(filename))  
-} else load(here(filename))
+ff <- bmf(kappa ~ 1 + setsize + (1 + setsize |subject),
+          thetat ~ 1 + setsize + (1 + setsize |subject))
+
+pr <- prior_('normal(0.0,0.8)', class = 'b', nlpar = 'kappa', lb = 0) +
+  prior_('logistic(0,1)', class = 'b', nlpar = 'thetat',lb = 0)
+
+filename <- here('output','fit_popov_unpublished_monotonic_constraints_2p')
+
+fit_popov_prior <- bmm(model = model_2p,
+                       formula = ff,
+                       data = data_popov,
+                       prior = pr,
+                       
+                       # save settings
+                       sample_prior = TRUE,
+                       save_pars = save_pars(all = TRUE),
+                       
+                       # add brms settings
+                       warmup = warmup_samples,
+                       iter = warmup_samples + postwarmup_samples, 
+                       chains = nChains,
+                       cores = parallel::detectCores(),
+                       
+                       # control commands for the sampler
+                       control = list(adapt_delta = adapt_delta, 
+                                      max_treedepth = max_treedepth),
+                       
+                       # save results to file
+                       file = filename)
```

Comment:
"Error: Object loaded via 'file' is not of class 'bmmfit'." - see my comment on Example 1